STYLE.md

```diff
@@ -0,0 +1,147 @@
+
+If a document proves a great deal longer or more structurally complex than those simplistic rules of thumb, there should be a good reason.
+
+## Hyperlink Considerations
+
+### Naming
+
+Name hyperlinks carefully to give them maximum context. For example, note that certain information is in the [style guide](STYLE.md), rather than just pointing lazily to the style guide [here](STYLE.md). The link text "here" gives almost no information about its target. It is helpful to [write a clear sentence](https://faculty.washington.edu/heagerty/Courses/b572/public/StrunkWhite.pdf) first, then bracket the choice words within to declare them a hyperlink.
+
+### Marking Down the Link
+
+As mentioned above, the inline style of hyperlinking is preferred to other ways of writing hyperlinks in Markdown. Using relative URLs where possible helps shorten these inline hyperlink declarations. In addition, relative links remain valid even as the site root moves. Absolute linking is obviously necessary for resources external to the document's repository and/or the cores.com domain.
+
+#### Example: Linking to the Quick Start Guide
+
+There are two ways to refer to the [quick start guide](os/quickstart.md):
+
+1. The preferred way is a relative link from the current file's path to the target: `[quick start guide](os/quickstart.md)`
+2. An absolute link is less flexible: `[quick start guide](https://github.com/coreos/docs/blob/master/os/quickstart.md)`
+
+
+#### Hyperlink Deployment Automation
+
+CoreOS documents have two major publication targets: the [coreos.com documentation library](https://coreos.com/docs/), and github's in-repo Markdown presentation. The deployment scripts used to build the CoreOS site handle some of the wrinkles arising between the two targets. These scripts expect links to other CoreOS project documentation to refer to the Markdown source, that is, to end with the `.md` file extension. The deployment scripts rewrite hyperlinks to replace that extension with `.html` for presentation. This allows the links to be valid in either context. External links are not rewritten.
+
+## Command Line Grammar
+
+### Terms
+
+*Commands* *invoke* or *execute* programs. Commands *take* *arguments* and *accept* *options*, which themselves may be *set* to *values*.
+
+### Example: Documenting `echo(1)`
+
+In this simple command line:
+
+```sh
+$ echo -n Example
+Example$
+```
+
+`echo` is the command, and `Example` is the argument. The option `-n` suppresses the terminating newline usually emitted by `echo`. A single-letter binary option like `-n` is sometimes called a *flag*. The `echo(1)` command prints its argument on the standard output, and a good shell excerpt often includes the expected output of commands, as shown here. The shell prompt character `$` distinguishes input from output.
+
+### Example: Documenting Subcommands
```

Comment:
Do we want to cover long commands here?

Do this:

```
$ docker run \
  -d \
  -p 80:80 \
  registry.example.com:5000/apache \
  /usr/sbin/apache2ctl -D FOREGROUND
```

instead of:

```
$ docker run -d -p 80:80 registry.example.com:5000/apache /usr/sbin/apache2ctl -D FOREGROUND
```